### PR TITLE
[specific ci=1-08-Docker-Logs] Fix streaming issue upon returning error in docker logs

### LINKED ISSUE #3682: Unrecognized input header: 69 on docker logs command during docker stop test
From build:  
https://ci.vmware.run/vmware/vic/7586

This appears to be the calls from docker-personality.log, but I don't see any particular errors here, I wonder if the error came from the client because it didn't like something we sent back?
```
time="2017-01-19T17:46:46Z" level=debug msg="Calling GET /v1.21/containers/f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615/logs?stderr=1&stdout=1&tail=all" 
time="2017-01-19T17:46:46Z" level=debug msg="Client and server don't have the same version (client: 1.11.2, server: 1.22)" 
time=2017-01-19T17:46:46.718247879Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerLogs:975] 
time=2017-01-19T17:46:46.718333453Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).StreamContainerLogs:425] 
time=2017-01-19T17:46:46.809507266Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).StreamContainerLogs:425] [91.167716ms]  
time=2017-01-19T17:46:46.809528511Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerLogs:975] [91.278576ms]  
time="2017-01-19T17:46:46Z" level=debug msg="Calling POST /v1.21/containers/f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615/stop?t=10" 
time="2017-01-19T17:46:46Z" level=debug msg="Client and server don't have the same version (client: 1.11.2, server: 1.22)" 
time=2017-01-19T17:46:46.856067542Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerStop:801] f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615 
time=2017-01-19T17:46:46.856081681Z level=debug msg=[BEGIN] [github.com/vmware/vic/pkg/retry.Do:38] 
time=2017-01-19T17:46:46.856138974Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).Stop:463] f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615 
time=2017-01-19T17:46:46.904854777Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).IsRunning:540] 
time=2017-01-19T17:46:46.916322364Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).Get:136] 
time=2017-01-19T17:46:46.916345967Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).getImageByDigest:176] 
time=2017-01-19T17:46:46.916352895Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).getImageByDigest:176] [5.479µs]  
time=2017-01-19T17:46:46.916358577Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).Get:136] [37.931µs]  
time="2017-01-19T17:46:46Z" level=debug msg="Docker inspect - network settings = &types.NetworkSettings{NetworkSettingsBase:types.NetworkSettingsBase{Bridge:\"\", SandboxID:\"\", HairpinMode:false, LinkLocalIPv6Address:\"\", LinkLocalIPv6PrefixLen:0, Ports:nat.PortMap{}, SandboxKey:\"\", SecondaryIPAddresses:[]network.Address(nil), SecondaryIPv6Addresses:[]network.Address(nil)}, DefaultNetworkSettings:types.DefaultNetworkSettings{EndpointID:\"\", Gateway:\"\", GlobalIPv6Address:\"\", GlobalIPv6PrefixLen:0, IPAddress:\"\", IPPrefixLen:0, IPv6Gateway:\"\", MacAddress:\"\"}, Networks:map[string]*network.EndpointSettings{\"bridge\":(*network.EndpointSettings)(0xc420820c00)}}" 
time=2017-01-19T17:46:46.916411847Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).IsRunning:540] [11.559032ms]  
time="2017-01-19T17:46:46Z" level=debug msg="UnmapPorts: map[]" 
time=2017-01-19T17:46:46.952348541Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).CommitContainerHandle:392] 3f52c441fd23dececaf037410b569a69 
time=2017-01-19T17:46:58.610086486Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).CommitContainerHandle:392] [11.657608744s] 3f52c441fd23dececaf037410b569a69 
time=2017-01-19T17:46:58.615634504Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).Stop:463] [11.759492181s] f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615 
time="2017-01-19T17:46:58Z" level=debug msg="Will not try again. Operation succeeded" 
time=2017-01-19T17:46:58.615730814Z level=debug msg=[ END ] [github.com/vmware/vic/pkg/retry.Do:38] [11.759647176s]  
time=2017-01-19T17:46:58.615911933Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerStop:801] [11.759669504s] f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615 
time="2017-01-19T17:46:58Z" level=debug msg="Calling GET /v1.21/containers/f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615/json" 
time="2017-01-19T17:46:58Z" level=debug msg="Client and server don't have the same version (client: 1.11.2, server: 1.22)" 
time=2017-01-19T17:46:58.666927872Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerInspect:910] f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615 
time="2017-01-19T17:46:58Z" level=debug msg="Found \"f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615\" in cache as \"f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615\"" 
time=2017-01-19T17:46:58.890571076Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).Get:136] 
time=2017-01-19T17:46:58.890621742Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).getImageByDigest:176] 
time=2017-01-19T17:46:58.890630898Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).getImageByDigest:176] [7.263µs]  
time=2017-01-19T17:46:58.890637522Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends/cache.(*ICache).Get:136] [67.918µs]  
time="2017-01-19T17:46:58Z" level=debug msg="Docker inspect - network settings = &types.NetworkSettings{NetworkSettingsBase:types.NetworkSettingsBase{Bridge:\"\", SandboxID:\"\", HairpinMode:false, LinkLocalIPv6Address:\"\", LinkLocalIPv6PrefixLen:0, Ports:nat.PortMap{}, SandboxKey:\"\", SecondaryIPAddresses:[]network.Address(nil), SecondaryIPv6Addresses:[]network.Address(nil)}, DefaultNetworkSettings:types.DefaultNetworkSettings{EndpointID:\"\", Gateway:\"\", GlobalIPv6Address:\"\", GlobalIPv6PrefixLen:0, IPAddress:\"\", IPPrefixLen:0, IPv6Gateway:\"\", MacAddress:\"\"}, Networks:map[string]*network.EndpointSettings{\"bridge\":(*network.EndpointSettings)(0xc4208212c0)}}" 
time="2017-01-19T17:46:58Z" level=debug msg="ContainerInspect json config = &{Hostname:f6de8b305041 Domainname: User: AttachStdin:false AttachStdout:false AttachStderr:false ExposedPorts:map[] PublishService: Tty:false OpenStdin:false StdinOnce:true Env:[PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin] Cmd:[sh -c trap 'echo StopSignalUSR1' USR1; echo READY; while true; do sleep 1; done] ArgsEscaped:false Image:busybox Volumes:map[] WorkingDir: Entrypoint:[] NetworkDisabled:false MacAddress: OnBuild:[] Labels:map[] StopSignal:USR1}\n" 
time="2017-01-19T17:46:58Z" level=debug msg="Docker inspect - network settings = &types.NetworkSettings{NetworkSettingsBase:types.NetworkSettingsBase{Bridge:\"\", SandboxID:\"\", HairpinMode:false, LinkLocalIPv6Address:\"\", LinkLocalIPv6PrefixLen:0, Ports:nat.PortMap{}, SandboxKey:\"\", SecondaryIPAddresses:[]network.Address(nil), SecondaryIPv6Addresses:[]network.Address(nil)}, DefaultNetworkSettings:types.DefaultNetworkSettings{EndpointID:\"\", Gateway:\"\", GlobalIPv6Address:\"\", GlobalIPv6PrefixLen:0, IPAddress:\"\", IPPrefixLen:0, IPv6Gateway:\"\", MacAddress:\"\"}, Networks:map[string]*network.EndpointSettings{\"bridge\":(*network.EndpointSettings)(0xc4208212c0)}}" 
time=2017-01-19T17:46:58.890820916Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerInspect:910] [223.892721ms] f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615 
time="2017-01-19T17:46:58Z" level=debug msg="Calling GET /v1.21/containers/f6de8b305041f839aa3cb238cd3deb38466be92ad0b621499b911342b95d4615/logs?stderr=1&stdout=1&tail=all" 
time="2017-01-19T17:46:58Z" level=debug msg="Client and server don't have the same version (client: 1.11.2, server: 1.22)" 
time=2017-01-19T17:46:58.949110874Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerLogs:975] 
time=2017-01-19T17:46:58.949199197Z level=debug msg=[BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).StreamContainerLogs:425] 
time=2017-01-19T17:46:58.950102231Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).StreamContainerLogs:425] [902.858µs]  
time=2017-01-19T17:46:58.950112102Z level=debug msg=[ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerLogs:975] [1.001585ms]  
```

[integration_logs_7586_358f257ce38d6d0565ceaa8b9728f48912a269b8.zip](https://github.com/vmware/vic/files/717654/integration_logs_7586_358f257ce38d6d0565ceaa8b9728f48912a269b8.zip)

### PR DESCRIPTION
**Pull request by anchal-agrawal**
Fixes #3682

### CODE REVIEW

**Comment by anchal-agrawal** [comment]
@mhagen-vmware [`StreamContainerLogs`](https://github.com/vmware/vic/blob/master/lib/apiservers/engine/backends/container_proxy.go/#L424) will return an error if there's a discrepancy between the PL and the persona's container caches, or if there's an error accessing the data from the datastore. Can we simulate such behavior in an integration test?

**Comment by sflxn** [comment]
@anchal-agrawal you can add additional cases to the containerlogs unit test in container_test.go.

File: lib/apiservers/portlayer/swagger.json [hunk 1]
```diff
@@ -1601,16 +1601,10 @@
 						}
 					},
 					"404": {
```
Comment (vburenin) [inline lib/apiservers/portlayer/swagger.json]: I wonder why did you remove schema from the error? Does docker interpret error code only?

**Comment by matthewavery** [comment]
lgtm, I think adding additional unit tests is a good call for this, it will help guard against accidental changes back in the future.

File: lib/apiservers/portlayer/swagger.json (hunk 1 above)
Comment (anchal-agrawal) [inline lib/apiservers/portlayer/swagger.json]: With the schema, we were getting this error from swagger while returning, for example, [this error](https://github.com/vmware/vic/blob/master/lib/apiservers/portlayer/restapi/handlers/containers_handlers.go#L343) from the portlayer:
```
Server error from portlayer: Unknown error from the interaction port layer: &{0 } (*models.Error) is not supported by the ByteStreamConsumer, can be resolved by supporting Writer/BinaryUnmarshaler interface
```

This was causing the [type switch on the error](https://github.com/vmware/vic/blob/master/lib/apiservers/engine/backends/container_proxy.go/#L439) to hit `default`.

**Comment by anchal-agrawal** [comment]
@sflxn @matthewavery I added a unit test case to cover the scenario when `StreamContainerLogs` returns an error and `ContainerLogs` doesn't.

**Comment by sflxn** [comment]
lgtm
